Test/TestPeptideWithSetMods.cs

```diff
@@ -34,6 +35,145 @@ public static void TestDifferentProteaseEquals()
             Assert.That(!pep1.GetHashCode().Equals(pep2.GetHashCode()));
         }
 
+        [Test]
+        public static void TestNonAndSemiSpecificDigests()
```

Comment:
Whoa, this will be really hard to inspect when we inevitably break this.

Please separate the setup logic and put it in a `[OneTimeSetup]` block, storing the `fiveCleavageProductsSemiTrypsin` in a private field in the test class.

Then, separate the different batches of tests and consider using `[TestCase()]` to test the different inputs to reuse more code.

Reply:
I don't understand the benefits of doing this. Can you explain whenever you get in?